hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2305,19 +2304,16 @@ public List<OzoneFileStatus> listStatus(String volumeName, String bucketName,
       String keyName, boolean recursive, String startKey,
       long numEntries, boolean allowPartialPrefixes) throws IOException {
     OmKeyArgs keyArgs = prepareOmKeyArgs(volumeName, bucketName, keyName);
-    return ozoneManagerClient
-        .listStatus(keyArgs, recursive, startKey, numEntries,
-            allowPartialPrefixes);
-  }
-
-  @Override
-  public List<OzoneFileStatusLight> listStatusLight(String volumeName,
-      String bucketName, String keyName, boolean recursive, String startKey,
-      long numEntries, boolean allowPartialPrefixes) throws IOException {
-    OmKeyArgs keyArgs = prepareOmKeyArgs(volumeName, bucketName, keyName);
-    return ozoneManagerClient
-        .listStatusLight(keyArgs, recursive, startKey, numEntries,
-            allowPartialPrefixes);
+    if (omVersion.compareTo(OzoneManagerVersion.LIGHTWEIGHT_LIST_STATUS) >= 0) {
```

Comment:
listStatus() is exposed interface via OzoneBucket for the client code, if some external has integrated, can impact the compatibility.
For newer client, if new interface is used and OM do not have that, it will return command not found. In that case, client can call old interface and convert output. This will provide better compatibility. And since this case is rare where client is new and server is old, this extra call for the duration should be enough.

Reply:
What is the actual exception corresponding to "command not found"? Do we get something that clean or is it just a general protobuf parsing error like the one shown in the PR description? We use versions to avoid having to make compatibility assumptions based on vague protobuf exceptions.

Reply:
@sumitagrawl we have not yet shipped the new API with a release, so it should be ok to pull this API. We should not have exposed an API which does not protect from version variation between client and OM. I would vote to remote this API.

Reply:
> Do we get something that clean or is it just a general protobuf parsing error like the one shown in the PR description?

We mainly get an `InvalidProtocolBufferException` with a "_Message missing required fields: cmdType_".

Reply:
> we have not yet shipped the new API with a release

I checked that we shipped both [#5115](https://github.com/apache/ozone/pull/5115) (lightweight listKeys API) as well as [#5275](https://github.com/apache/ozone/pull/5275) (lightweight listStatus API) as part of 1.4.0 ([ozone-1.4.0 tag](https://github.com/apache/ozone/tree/ozone-1.4.0)).

Reply:
@kerneltime with this implementation, we are changing behavior of OzoneBucket --> listStatus()
If this api is integrated to any external service, can have impact.

It should be only new API changes.

Reply:
Thank you for the review, please find the simplified patch: https://github.com/tanvipenumudy/ozone/commit/7d82fb6f2d10833b4e16af1008db9573cca245bc